### DISCUSSION TITLE
load checkpoint resume from CLI, different learning rate

### DISCUSSION THREAD

**Discussion by mshooter** [Lightning-AI/pytorch-lightning#18339] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi how do I load the weights of the lightning module? but change learning rate using CLI ?

**Comment by mauvilsa** [comment]
This depends on the modules and CLI you implemented. If you implemented the minimal `LightningCLI` with subcommands and the automatic configure_optimizers, the it could be:

```shell
# Initial run
python cli.py fit [some settings] --optimizer SGD --optimizer.lr 0.01
# second run
python cli.py fit [other settings] --optimizer SGD --optimizer.lr 0.05 --ckpt_path lightning_logs/.../checkpoints/[some name].ckpt
```

**Reply by mshooter** [reply]
so I have a parameter self.learning_rate which I normally change. so I have to set the --optimizer.lr directly for it to work? what about the schedulers?

**Reply by mauvilsa** [reply, marked as answer]
It depends on what you implement. What `LightningCLI` does is expose the parameters in `__init__` as configurable from command line and config files. By `self.learning_rate` do you mean that you added an init parameter for the learning rate?

Note that the CLIs have a help that explains how to use it, i.e. `python cli.py --help`. Also, look at the CLI documentation. The use of the automatic `configure_optimizers`, both for optimizers and schedulers is explained [here](https://lightning.ai/docs/pytorch/stable/cli/lightning_cli_intermediate_2.html#multiple-optimizers). Though, if you have defined a learning rate parameter then it seems you are not doing that. A more advanced way of making optimizers and schedulers configurable (via dependency injection) is explained in [multiple-optimizers-and-schedulers](https://lightning.ai/docs/pytorch/stable/cli/lightning_cli_advanced_3.html#multiple-optimizers-and-schedulers).
